### DISCUSSION TITLE
Simplify the constructor of a derived class

### DISCUSSION THREAD

**Discussion by skyoxZ** [dotnet/csharplang#4277] [Language Ideas]
Suppose such a class:

    public class KnownException : Exception
    {
        public string Code { get; }

        /// <param name="message">Error message.</param>
        /// <param name="code">Error code.</param>
        /// <param name="innerException">Inner exception.</param>
        public KnownException(string message, string code, Exception innerException)
            : base(message, innerException)
            => Code = code;
    }

It would be great if the constructor can be defined as:

        /// <param name="code">Error code.</param>
        public KnownException(string, string code, Exception) => Code = code;

The name and comment of param is inherited.

This feature is also useful in asp.net core if I have such a base service class:

    public abstract class BaseService
    {
        protected readonly ILogger _logger;

        protected readonly MainDbContext _db;

        protected BaseService(ILoggerFactory loggerFactory, MainDbContext db)
        {
            _logger = loggerFactory.CreateLogger(GetType());
            _db = db;
        }
    }

**Comment by TonyValenti** [comment]
This is a great idea!
It would also be nice if a derived class that defined no constructors would automatically synthesize constructors that match all child constructors.

**Comment by huoyaoyuan** [comment]
The language should be careful about synthesizing constructors based on **order and name**.

**Reply by 333fred** [reply]
Yes. This is a good example of why we're looking at required properties: you don't care about order for such properties, you do for constructor parameters.
